sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/QueueReceiverAsyncClient.java

```diff
@@ -0,0 +1,238 @@
+     *
+     * @return A stream of events for every partition from Queue.
+     *
+     * @throws NullPointerException if {@code receiveMode} is null.
+     */
+    public Flux<Message> receive(ReceiveMode receiveMode) {
+        if (Objects.isNull(receiveMode)) {
+            return fluxError(logger, new NullPointerException("'receiveMode' cannot be null."));
+        }
+
+        final String linkName = connectionProcessor.getEntityPath();
+        return createConsumer(linkName, receiveMode);
+    }
+
+    /**
+     * Disposes of the consumer by closing the underlying connection to the service.
+     */
+    @Override
+    public void close() {
+        if (isDisposed.getAndSet(true)) {
+            return;
+        }
+        if (openConsumer.get() != null) {
+            openConsumer.get().close();
+        }
+
+
+        connectionProcessor.dispose();
+
+    }
+
+    private Flux<Message> createConsumer(String linkName, ReceiveMode receiveMode) {
+        if (openConsumer.get() == null) {
+            logger.info("{}: Creating receive consumer.", linkName);
+            openConsumer.set(createServiceBusConsumer(linkName, receiveMode));
+        }
+        return openConsumer.get()
+            .receive()
+            .doOnCancel(() -> removeLink(linkName, SignalType.CANCEL))
+            .doOnComplete(() -> removeLink(linkName, SignalType.ON_COMPLETE))
+            .doOnError(error -> removeLink(linkName, SignalType.ON_ERROR));
+    }
+
+    private void removeLink(String linkName, SignalType signalType) {
+        logger.info("{}: Receiving completed. Signal[{}]", linkName, signalType);
+
+        if (openConsumer.get() != null) {
+            openConsumer.get().close();
+        }
+    }
+    private ServiceBusAsyncConsumer createServiceBusConsumer(String linkName, ReceiveMode receiveMode) {
+        final String entityPath = String.format(Locale.US, RECEIVER_ENTITY_PATH_FORMAT, getQueueName());
+
+        final Flux<AmqpReceiveLink> receiveLinkMono =
+            connectionProcessor.flatMap(connection ->
+                connection.createReceiveLink(linkName, entityPath, receiveMode))
+                .doOnNext(next -> logger.verbose("Creating consumer for path: {}", next.getEntityPath()))
+                .repeat();
+
+        final AmqpRetryPolicy retryPolicy = RetryUtil.getRetryPolicy(connectionProcessor.getRetryOptions());
+        final ServiceBusReceiveLinkProcessor linkMessageProcessor = receiveLinkMono.subscribeWith(
+            new ServiceBusReceiveLinkProcessor(prefetchCount, retryPolicy, connectionProcessor));
+
+        return new ServiceBusAsyncConsumer(linkMessageProcessor, messageSerializer, fullyQualifiedNamespace,
+            entityPath);
+    }
+
+    /**
+     *
+     * @param lockToken to be used.
+     * @return The {@link Mono} the finishes this operation on service bus resource.
+     */
+    public Mono<Void> abandon(UUID lockToken) {
+        //TODO(feature-to-implement)
+        return null;
```

Comment:
Imho, returning Mono.empty() is more friendly than returning null for writing API tests before implementation.